### ISSUE TITLE
Module requires react types to be installed

### ISSUE THREAD

**Issue by mihkeleidast** [TroyAlford/react-jsx-parser#164]
When installing this module, I get the following warnings from npm:
```
npm WARN react-jsx-parser@1.27.0 requires a peer of @types/react@>=16 but none is installed. You must install peer dependencies yourself.
npm WARN react-jsx-parser@1.27.0 requires a peer of @types/react-dom@>=16 but none is installed. You must install peer dependencies yourself.
```
This should be not necessary when a project does not use TypeScript. When a React project does use TypeScript, it will almost definitely already have react types installed.

Can the types be removed from peerDependencies?

**Comment by TroyAlford** [comment]
The warning shouldn't break anything, and can be safely ignored. The reason I added it is that without that peer dep being installed, the TS types for this lib do not work. It seemed better, in my opinion, to make a non-breaking warning appear for folks who _don't_ need it than to give _no_ warning/explanation about why the typings don't work for folks who _do_.

Given that I've had 3 separate issues in the past couple months about making the TS types work consistently, I may be a bit over-sensitive about ensuring they're functional, though. I'll leave this issue open for a bit to see if others chime in supporting one way or the other.

**Comment by mihkeleidast** [comment]
I personally follow packages' peerDependency requirements very strictly to avoid issues (and I'm not the only one I hope), so ignoring these warnings is not an option to me. If you say I must install react types in order to use this package I will, but I know they are not necessary for everyone, hence this issue.

**Comment by TroyAlford** [comment]
I moved these from `peerDependencies` to `optionalDependencies`. This is now released in [1.28.1](https://github.com/TroyAlford/react-jsx-parser/releases/tag/1.28.1)
